### ISSUE TITLE
Relational Filter

### ISSUE THREAD

**Issue by wwwhttpru** [apstanisic/directus-dart#36]
I observe a lack of implementation for relational Filter.
How can I implement this with the current filter?
```
{
	"author": {
		"name": {
			"_eq": "Rijk van Zanten"
		}
	}
}
```

**Comment by wwwhttpru** [comment]
My solution to the problem was the following:
add in filter.dart
```
  const Filter.child(String value, Filter filter)
      : value = filter,
        comparison = value;
```
and
change toMapEntry method
```
  MapEntry<String, Object?> toMapEntry(String field) {
    // If value is a list of non filters
    if (value is List<Map<String, Filter>>) {
      if (comparison == '_or' || comparison == '_and') {
        return MapEntry(
          comparison,
          convertFilterList(value! as List<Map<String, Filter>>),
        );
      }
    }

    if (value != null && value is Filter) {
      final filter = value! as Filter;
      final childMap = filter.toMapEntry(comparison);
      return MapEntry(field, {
        childMap.key: childMap.value,
      });
    }

    return MapEntry(field, {comparison: value});
  }
```

**Comment by wwwhttpru** [comment]
Вот пример реализации в коде
```
final f = Filters(
      <String, Filter>{
        'author': Filter.child(
          'name',
          Filter.eq('Rijk van Zanten'),
        ),
      },
    );
```

result: f.toMap();

```
{author: {name: {_eq: Rijk van Zanten}}}
```

**Comment by apstanisic** [comment]
Maybe we can change `Map<String, Filter> data;` to `Map<String, object>`
https://github.com/apstanisic/directus-dart/blob/17ce3c8dee89422620913999422e4c318af94e24/lib/src/data_classes/filters.dart#L29

Only problem is that with that way we lose some type safety, but user could then pass simple JSON.
Do you think it's to verbose having to type `Filter.xxx` for every comparison?

**Comment by wwwhttpru** [comment]
I have used my own class before and I don't think it's very safe.
I don't see any difficulty in writing Filter.xxx
```
abstract class ApiFilter {
  ApiFilter._();

  static Map<String, Object?> filterItem(
    String superFiled,
    Object items,
  ) {
    return <String, Object?>{
      superFiled: items,
    };
  }

  static Map<String, Object?> filterItems(List<Map<String, Object?>> items) {
    final map = <String, Object?>{};
    for (final item in items) {
      map.addAll(item);
    }
    return map;
  }

  /// Равно
  static const String eq = '_eq';

  /// Не равно
  static const String neq = '_neq';

  /// Меньше
  static const String lt = '_lt';

  /// Меньше или равно
  static const String lte = '_lte';
  ...
}
```

I am ready to consider different implementations, but I would like to do so that would not spoil the current api

**Comment by apstanisic** [comment]
Yeah, if you can create PR without breaking change, that would be awesome

**Comment by apstanisic** [comment]
Closed in #37
